debezium-core/src/main/java/io/debezium/relational/TableIdParser.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.relational;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.debezium.text.ParsingException;
+import io.debezium.text.TokenStream;
+import io.debezium.text.TokenStream.CharacterStream;
+import io.debezium.text.TokenStream.Tokenizer;
+import io.debezium.text.TokenStream.Tokens;
+
+/**
+ * Parses identifiers into the corresponding parts of a {@link TableId}.
+ *
+ * @author Gunnar Morling.
+ */
+class TableIdParser {
+
+    private static final char SEPARATOR = '.';
+
+    public static List<String> parse(String identifier, char quotingChar) {
+        TokenStream stream = new TokenStream(identifier, new TableIdTokenizer(identifier, quotingChar), true);
+        stream.start();
+
+        List<String> parts = new ArrayList<>();
+
+        while(stream.hasNext()) {
+            parts.add(stream.consume());
+        }
+
+        return parts;
+    }
+
+    private static class TableIdTokenizer implements Tokenizer {
```

Comment:
Seems like it would have been easier to fix in BasicTokenizer rather than to come up with completely different parsing logic. I guess I have to admit I'm not a fan of doing this a completely different way than the other tokenizers. For example, simply changing [these lines in BasicTokenizer](https://github.com/debezium/debezium/blob/master/debezium-core/src/main/java/io/debezium/text/TokenStream.java#L2457-L2460):

    } else if (c == '"') {
        foundClosingQuote = true;
        break;
    }

to this:

    } else if (c == '"') {
        if (input.isNext('"')) {
            // The quote character is escaped by two in a row
            c = input.next();
        } else {
            foundClosingQuote = true;
            break;
        }
    }

here and in a few other places would have supported escaping quote characters with two quote characters.